### ISSUE TITLE
Jackson auto configuration must not be triggered if spring-web is not on the classpath

### ISSUE THREAD

**Issue by olivergierke** [spring-projects/spring-boot#10166]
Using the JSON starter pulls Jackson in the classpath, which in turn triggers Jackson auto-configuration trying to use `Jackson2ObjectMapperBuilder` from `spring-web` which is not in the classpath necessarily. This causes the auto-configuration to fail.

A scenario that pulls in the JSON starter but nothing web related in the first place might be a messaging app that intends to use JSON as the message serialization format.

**Comment by wilkinsona** [comment]
@olivergierke Can you please share some configuration that reproduced the failure you saw? Looking at `JacksonAutoConfiguration` the necessary conditions for requiring `Jackson2ObjectMapperBuilder` appear to be in place already.

**Comment by olivergierke** [comment]
On close inspection of the problem I think the situation is slightly different. I've got a project depending on `spring-boot-starter-json` and then want to consume an `ObjectMapper` through an `@Bean`-method. I get the following error:

```
***************************
APPLICATION FAILED TO START
***************************

Description:

Parameter 0 of method projectionFactory in example.sos.messaging.inventory.integration.MessagingConfiguration required a bean of type 'com.fasterxml.jackson.databind.ObjectMapper' that could not be found.
	- Bean method 'jacksonObjectMapper' not loaded because @ConditionalOnClass did not find required class 'org.springframework.http.converter.json.Jackson2ObjectMapperBuilder'


Action:

Consider revisiting the conditions above or defining a bean of type 'com.fasterxml.jackson.databind.ObjectMapper' in your configuration.
```

I guess the second part of the error message is where my assessment came from. I guess my naive assumption was that I could just depend on the JSON starter and would get an `ObjectMapper` bean in the application context that can be configured through Boot means (properties, module beans etc.) which doesn't seem to be the case really.

Maybe with #10167 fixed, we can just drop this one? An alternative would be to create a simple `ObjectMapper` in case the builder is not present but I guess that'd would then cause some customizations to work, but not all of them.

**Comment by wilkinsona** [comment]
Thanks, @olivergierke. So what's happened there is that the lack of `spring-web` on the classpath has caused the `ObjectMapper` auto-configuration to back off. That's fixed by #10031.

> An alternative would be to create a simple ObjectMapper in case the builder is not present but I guess that'd would then cause some customizations to work, but not all of them.

Yes, I think it would and that would be confusing. You'd end up with an `ObjectMapper` that did not reflect all of the `spring.jackson.*` properties, for example.

In summary, I think we should consider this to be a duplicate of #10031 as well.
